### PR TITLE
GPS API Documentation

### PR DESCRIPTION
**Pull request by WillXuCodes**
Added the new GPS API Documentation, please review and let me know if there are any mistakes or enhancements to make.

### CODE REVIEW

File: v5/api/c/gps.rst [hunk 1]
```diff
@@ -0,0 +1,681 @@
...
+
+=========
+GPS C API
+=========
+
+.. note:: For a pros-specific usage guide on the GPS, please check out our article
+          `here <..\tutorials\topical\gps.rst>`_.
+
+.. note::
+   For the C++ API, check out it `here<../cpp/gps.rst>`_.
```
Comment (kunwarsahni01) [inline v5/api/c/gps.rst:12]: I don't think this is needed.

File: v5/api/c/gps.rst [hunk 2]
```diff
@@ -0,0 +1,681 @@
...
+ Value
+================================== =====================================================================================
+x                                   X Position (meters)
+y                                   Y Position (meters)
+pitch                               Percieved Pitch based on GPS and IMU
+roll                                Percieved Roll based on GPS and IMU
+yaw                                 Percieved Yaw based on GPS and IMU
+================================== =====================================================================================
+
+---
```
Comment (kunwarsahni01) [inline v5/api/c/gps.rst:643]: I think this is causing some weird lines in the produced files, applies to the C++ api as well.

File: v5/api/c/gps.rst [hunk 3]
```diff
@@ -0,0 +1,681 @@
...
+
+::
+
+  struct gps_raw_s {
+	double x;
+	double y;
+	double z;
+  };
+
+---
```
Comment (kunwarsahni01) [inline v5/api/c/gps.rst:656]: Same here: 
Screen Shot 2021-08-14 at 2 20 24 PM

File: v5/api/c/gps.rst [hunk 4]
```diff
@@ -0,0 +1,681 @@
...
+gps_initialize_full
+-------------------
+
+Set the GPS's offset relative to the center of turning in meters, as well as its initial position.
+
+This function uses the following values of ``errno`` when an error state is reached:
+
+- ``ENXIO`` - The given value is not within the range of V5 ports (1-21).
+- ``ENODEV`` - The port cannot be configured as a GPS.
+- ``EAGAIN``` - The sensor is still calibrating.
```
Comment (kunwarsahni01) [inline v5/api/c/gps.rst:28]:
```suggestion
- ``EAGAIN`` - The sensor is still calibrating.
```
Should be changes in all other instances as well

File: v5/api/c/gps.rst [hunk 5]
```diff
@@ -0,0 +1,681 @@
...
+
+.. tabs ::
+   .. tab :: Prototype
+      .. highlight:: c
+      ::
+
+         int32_t gps_initialize_full(uint8_t port, double xInitial, double yInitial, double headingInitial, double xOffset,
```
Comment (kunwarsahni01) [inline v5/api/c/gps.rst:35]: Prototype is not getting syntax highlighting:
Screen Shot 2021-08-14 at 2 23 14 PM

File: v5/api/c/gps.rst [hunk 6]
```diff
@@ -0,0 +1,681 @@
...
+      .. highlight:: c
+      ::
+        #define GPS_PORT 1
+        #define X_OFFSET 0
+        #define Y_OFFSET 0
+        #define X_INITIAL 0
+        #define Y_INITIAL 0
+        #define HEADING_INITIAL 0
+
+        void initialize() {
```
Comment (kunwarsahni01) [inline v5/api/c/gps.rst:48]: Example not rendering:
Screen Shot 2021-08-14 at 2 24 19 PM

**Review by kunwarsahni01 — CHANGES_REQUESTED** [review]

File: v5/api/c/gps.rst (hunk 2 above)
Comment (WillXuCodes) [reply v5/api/c/gps.rst:653]: It seems this was intentional, but is inconsistent throughout the docs whether it be a short line or a long line. I'm gonna make the GPS's consistent at the very least by making them long lines.

File: v5/api/c/gps.rst (hunk 5 above)
Comment (WillXuCodes) [reply v5/api/c/gps.rst:32]: Gonna be honest, I don't see how it isn't getting proper syntax highlighting in the example above. It seems consistent with the rest of the docs.

File: v5/api/c/gps.rst (hunk 6 above)
Comment (WillXuCodes) [reply v5/api/c/gps.rst:46]: Should be fixed

image

Comment (WillXuCodes) [reply v5/api/c/gps.rst:46]: Though we should change these not to be 0s and 1s...

**Review by kunwarsahni01 — APPROVED** [review]
LGTM

**Pull request merged**